### DISCUSSION TITLE
[graphiql] Make graphiql explorer nodes collapsible without unchecking.

### DISCUSSION THREAD

**Discussion by acalvino4** [graphql/graphiql#2955] [Q&A / Support]
The graphql explorer sidebar gets extremely long to scroll through very fast as you select various data to query. This is because in checking a box it expands to show all other options you could could potentially select. On can collapse the view by clicking on the checked entry again, but this deselects it. It would be awesome if we could collapse those segments of the tree without deselecting them.

Reference screenshot:
![image](https://user-images.githubusercontent.com/43254861/206000637-08e240ce-fdc7-422d-b128-d337a4de1137.png)

**Comment by jonathanawesome** [comment]
Hey there @acalvino4, thanks for the feedback and suggestion!

The bit that you're describing is actually not part of GraphiQL core, it's the OneGraph Explorer [plugin that was built](https://github.com/OneGraph/graphiql-explorer) for GraphiQL < V2 and, thanks to @thomasheyenbrock, is included as a v2 plugin in this repository as a "legacy" tool so that the community that has moved onto GraphiQL V2 is not void of the feature set while we work on a new, official plugin with this functionality. 

Unfortunately, OneGraph's Explorer repository seems mostly abandoned so getting your suggestion into that codebase is unlikely.

That said...your suggestion is excellent, arbitrarily expanding and collapsing to improve the UX, and is included in the early designs for this new plugin. [Here's a very lengthy and informative discussion](https://github.com/graphql/graphiql/discussions/2216) about this feature and others. And here's [another discussion](https://github.com/graphql/graphiql/discussions/2788) that I started back in October that's specific to the new visual operation builder. Would appreciate your thoughts and feedback there as well!
